server/graphql/v1/mutations/orders.js

```diff
@@ -318,14 +321,27 @@ export async function createOrder(order, loaders, remoteUser, reqIp) {
       throw new Error(`Invalid currency. Expected ${currency}.`);
     }
 
-    const quantity = order.quantity || 1;
+    // ---- Checks on totalAmount ----
+    if (order.totalAmount < 0) {
+      throw new Error('Total amount cannot be a negative value');
+    }
 
-    let totalAmount;
-    if (tier && tier.amount && !tier.presets) {
-      // if the tier has presets, we can't enforce tier.amount
-      totalAmount = tier.amount * quantity;
-    } else {
-      totalAmount = order.totalAmount; // e.g. the donor tier doesn't set an amount
+    // Don't allow custom values if using a tier with fixed amount
```

Comment:
@xdamman can you confirm we want that?

Reply:
@xdamman This behaviour is better explained in https://github.com/opencollective/opencollective/issues/1637

Reply:
Something to note on this one: we were already forcing the amount. The change is to throw an error instead of forcing it on the API, this should be enforced on frontend, not on the API.

The idea is that if the user send a total amount of $40, we should never bill more or less than this amount. If it doesn't match, then it's an error.

Reply:
So this was breaking some tests, apparently parts of the code don't send the `totalAmount` if using a fixed tier. I've made a change to only throw an error if `totalAmount` is not null or undefined.

See https://github.com/opencollective/opencollective-api/pull/1671/commits/59a176d9d41e486c149bc7f06381e2577e87645b

Reply:
Yes I confirm.